src/App.js

```diff
@@ -0,0 +1,109 @@
+import { Component } from 'react';
+import AppHeader from './AppHeader';
+import ItemsList from './ItemsList';
+import AppFooter from './AppFooter';
+
+class App extends Component {
+    constructor() {
+        super()
+        this.state = {
+            inputfield: '',
+            inputs: [],
+            saveButtonClassActive: false
+        };
+    }
+
+    componentDidMount() {
+        const storageItems = localStorage.getItem('items');
+        const parsedItems = JSON.parse(storageItems);
+        if (parsedItems) {
+            const items = [];
+            for (const item of parsedItems) {
+                const input = {
+                    value: item.value,
+                    isChecked: item.isChecked
+                };
+                items.push(input);
+            }
+            this.setState({ inputs: items });
+        }
+    }
+
+    onInputChange = event => {
+        this.setState({ inputfield: event.target.value })
+    }
+
+    onAddButtonClick = () => {
+        if (this.state.inputfield) {
+            this.setState(state => {
```

Comment:
why are you using here function for setting a state?

I believe that this state update can be done without a state update function. Without this function, code would look a little bit simpler which makes this option as the preferred one.

Reply:
You're right. I can do this without the function as well.